packages/cli/src/utils/installationInfo.ts

```diff
@@ -0,0 +1,174 @@
+  }
+
+  try {
+    const realPath = fs.realpathSync(cliPath);
+    const isGit = isGitRepository(process.cwd());
+
+    // Check for local git clone first
+    if (
+      isGit &&
+      projectRoot &&
+      realPath.startsWith(projectRoot) &&
+      !realPath.includes('node_modules')
+    ) {
+      return {
+        packageManager: PackageManager.UNKNOWN, // Not managed by a package manager in this sense
+        isGlobal: false,
+        updateMessage:
+          'Running from a local git clone. Please update with "git pull".',
+      };
+    }
+
+    // Check for npx/pnpx/bunx
+    if (
+      realPath.includes(path.join('.npm', '_npx')) ||
+      realPath.includes(path.join('npm', '_npx'))
+    ) {
+      return {
+        packageManager: PackageManager.NPX,
+        isGlobal: false,
+        updateMessage: 'Running via npx, update not applicable.',
+      };
+    }
+    if (realPath.includes(path.join('.pnpm', '_pnpx'))) {
+      return {
+        packageManager: PackageManager.NPX,
+        isGlobal: false,
+        updateMessage: 'Running via pnpx, update not applicable.',
+      };
+    }
+    if (realPath.includes(path.join('.bun', 'install', 'cache'))) {
+      return {
+        packageManager: PackageManager.NPX,
+        isGlobal: false,
+        updateMessage: 'Running via bunx, update not applicable.',
+      };
+    }
+
+    // Check for Homebrew
+    try {
+      // The package name in homebrew is gemini-cli
+      childProcess.execSync('brew list -1 | grep -q "^gemini-cli$"', {
+        stdio: 'ignore',
+      });
+      return {
+        packageManager: PackageManager.HOMEBREW,
+        isGlobal: true,
+        updateMessage:
+          'Installed via Homebrew. Please update with "brew upgrade".',
+      };
+    } catch (_error) {
+      // Brew is not installed or gemini-cli is not installed via brew.
+      // Continue to the next check.
+    }
+
+    // Check for pnpm
+    if (realPath.includes(path.join('.pnpm', 'global'))) {
+      const updateCommand = 'pnpm add -g @google/gemini-cli@latest';
+      return {
+        packageManager: PackageManager.PNPM,
+        isGlobal: true,
+        updateCommand,
+        updateMessage: isAutoUpdateDisabled
+          ? `Please run ${updateCommand} to update`
+          : 'Installed with pnpm. Attempting to automatically update now...',
+      };
+    }
+
+    // Check for yarn
+    if (realPath.includes(path.join('.yarn', 'global'))) {
+      const updateCommand = 'yarn global add @google/gemini-cli@latest';
+      return {
+        packageManager: PackageManager.YARN,
+        isGlobal: true,
+        updateCommand,
+        updateMessage: isAutoUpdateDisabled
+          ? `Please run ${updateCommand} to update`
+          : 'Installed with yarn. Attempting to automatically update now...',
+      };
+    }
+
+    // Check for bun
+    if (realPath.includes(path.join('.bun', 'bin'))) {
+      const updateCommand = 'bun add -g @google/gemini-cli@latest';
```

Comment:
wait a second bun entry? can they be merged?

Reply:
i moved it down to be near the other bun check, but bunx is different then bun since they install to different places. Bunx is similar to NPX where it doesn't actually install it and just runs it there, so we would need both checks but i could be wrong.